Realm.Shared/Realm.cs

```diff
@@ -718,7 +736,7 @@ public void RemoveAll()
 
             foreach (var @object in Schema)
```

Comment:
This could be rewritten to foreach the metadata dictionary, which would make it more efficient.